packages/bundlers/default/src/DefaultBundler.js

```diff
@@ -112,6 +112,12 @@ export default new Bundler({
     // Step 2: remove assets that are duplicated in a parent bundle
     bundleGraph.traverseBundles({
       exit(bundle) {
+        if (bundle.env.isIsolated()) {
```

Comment:
This could be done on an asset level, but checking the bundle allows us to skip this entire traversal.